### PR TITLE
Unfinalized Support / Simplied `GatewayRequest`

### PR DESCRIPTION
**Pull request by adraffy**
* Unfinalized OPFault
    * `OPFaultRollup` / `OPFaultVerifier` have `minAgeSec`
    * new `OPFaultGameFinder` with `minAgeSec`
        * `0` = finalized, otherwise required age of `DisputeGame`
* Unfinalized Linea
    * `UnfinalizedLineaRollup` reuses `LineaConfig` and adds `minAgeBlocks`
         * separate implementation since commit logic is divergent
    * Note: `UnfinalizedLineaVerifier` cannot prove commit time yet
    * Note:  `linea_getProof` doesn't generate unfinalized proofs yet
    * works when tested with an insane (2 day) delay
* Unfinalized Nitro
    * `NitroRollup` / `NitroGateway` have `minAgeBlocks`
* Delayed OP
    * `OPVerifier` has `minAgeSec`
    * `OPRollup` is unchanged
    * `Gateway(OPRollup)` can serve both finalized and delayed with sufficiently large `commitDepth`
* changed `GatewayRequest`
    * removed `inputs[]` and `defineInput*()`
    * significantly smaller program size
    * added missing solidity functions
* fixed bug when `NitroVerifier` is supplied a node not in the finalized chain
* `OP_NONZERO` &rarr; `OP_ZERO`
* `OP_EVAL_LOOP`
    * fixed bug (off by 1) when early-terminated
    * `ACQUIRE_STATE` now pushes remaining stack instead of replacing
    * `ACQUIRE_STATE` only executes when terminated early
    * new `KEEP_ARGS` keeps unvisited args on the stack when early terminated
* `OP_PUSH_INPUT` repurposed to `OP_PUSH_STACK` (same as `OP_DUP` but from the bottom)
    * possibly these could both be renamed to be similar
* rename `OPReverseRollup` &rarr; `ReverseOPRollup` 
    * standard prefixes: `Reverse-`, `Self-` and `Unfinalized-`
* fixed uncaught exception with `LRU.setPending()` &rarr; `setFuture()`

### CODE REVIEW

File: contracts/AbstractVerifier.sol [hunk 1]
```diff
@@ -35,8 +35,12 @@ abstract contract AbstractVerifier is IGatewayVerifier, Ownable {
         return _window;
     }
 
+    function getHooks() external view returns (IVerifierHooks) {
```
Comment (clowestab) [inline contracts/AbstractVerifier.sol:38]: Whats the intention of exposing this?

Comment (adraffy) [reply contracts/AbstractVerifier.sol:38]: figured it might be useful 🤷‍♂️

File: test/gateway/delayed-base.test.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { OPRollup } from '../../src/op/OPRollup.js';
+import { testOP } from './common.js';
+
+testOP(OPRollup.baseMainnetConfig, {
+  // https://basescan.org/address/0x0C49361E151BC79899A9DD31B8B0CCdE4F6fd2f6
+  slotDataContract: '0x0C49361E151BC79899A9DD31B8B0CCdE4F6fd2f6',
+  // delay by 1 hour
+  // NOTE: to delay longer, Gateway.commitDepth needs to be bigger
+  minAgeSec: 3600,
```
Comment (clowestab) [inline test/gateway/delayed-base.test.ts:9]: For Base we are requesting historical even though finalised data does exist I.E saying get/proof state from last week whereas with OPFault we say '6 hours ago is OK', there isnt finalised data?

Comment (adraffy) [reply test/gateway/delayed-base.test.ts:9]:
```
       T=0         1hr     ...     6hr       ...      3.5day
OP      |       Finalized  ==>   Delayed
OPFault |                      Unfinalized   <==    Finalized
```

**Pull request merged**